src/openloco/windows/build_vehicle.cpp

```diff
@@ -1,27 +1,285 @@
+    static loco_global<int32_t, 0x011364E8> _buildTargetVehicle; // -1 for no target
+    static loco_global<uint32_t, 0x011364EC> _numTrackTypeTabs;
+    // Array of types if 0xFF then no type, flag (1<<7) as well
+    static loco_global<int8_t[widxToTrackTypeTab(widx::tab_track_type_7) + 1], 0x011364F0> _TrackTypesForTab;
+    static std::array<uint16_t, 6> _scrollRowHeight = { { 22, 22, 22, 22, 42, 30 } };
+
+    loco_global<uint16_t[8], 0x112C826> _common_format_args;
+    static loco_global<string_id, 0x009C68E8> gGameCommandErrorTitle;
+    static loco_global<uint32_t[32], 0x00525E5E> currencyMultiplicationFactor;
+    static loco_global<uint8_t, 0x00525FC5> _525FC5;
+    static loco_global<uint8_t, 0x00525FAA> last_railroad_option;
+    static loco_global<uint8_t, 0x00525FAB> last_road_option;
+    static loco_global<uint8_t, 0x0052622C> last_build_vehicles_option; // Type is VehicleType
+    static loco_global<uint16_t, 0x0052622E> _52622E;                   // Tick related
+
+    static window_event_list _events;
 
     static void sub_4B92A5(ui::window* window);
+    static void setDisabledTransportTabs(ui::window* window);
+    static void setTrackTypeTabs(ui::window* window);
+    static void resetTrackTypeTabSelection(ui::window* window);
+    static void setTopToolbarLastTrack(uint8_t trackType, bool isRoad);
+    static void setTransportTypeTabs(ui::window* window);
+    static void sub_4B60CC(openloco::vehicle* vehicle);
+    static void drawVehicleOverview(gfx::drawpixelinfo_t* dpi, int16_t vehicleTypeIdx, company_id_t company, uint8_t eax, uint8_t esi, gfx::point_t offset);
+    static int16_t drawVehicleInline(gfx::drawpixelinfo_t* dpi, int16_t vehicleTypeIdx, uint8_t unk_1, company_id_t company, gfx::point_t loc);
+    static void drawTransportTypeTabs(ui::window* window, gfx::drawpixelinfo_t* dpi);
+    static void drawTrackTypeTabs(ui::window* window, gfx::drawpixelinfo_t* dpi);
+
+    static void on_mouse_up(ui::window* window, widget_index widgetIndex);
+    static void on_resize(window* window);
+    static void on_periodic_update(window* window);
+    static void on_update(window* window);
+    static void get_scroll_size(ui::window* window, uint32_t scrollIndex, uint16_t* scrollWidth, uint16_t* scrollHeight);
+    static void on_scroll_mouse_down(ui::window* window, int16_t x, int16_t y, uint8_t scroll_index);
+    static void on_scroll_mouse_over(ui::window* window, int16_t x, int16_t y, uint8_t scroll_index);
+    static void tooltip(FormatArguments& args, ui::window* window, widget_index widgetIndex);
+    static ui::cursor_id cursor(window* window, int16_t widgetIdx, int16_t xPos, int16_t yPos, ui::cursor_id fallback);
+    static void prepare_draw(ui::window* window);
+    static void draw(ui::window* window, gfx::drawpixelinfo_t* dpi);
+    static void draw_scroll(ui::window* window, gfx::drawpixelinfo_t* dpi, uint32_t scrollIndex);
+
+    static void initEvents()
```

Comment:
I generally put this function at the bottom of the namespace, so the event functions do not have to be predeclared. It's not something that needs changing, per se, but something to keep in mind, perhaps.